src/TheThingsNetwork.h

```diff
@@ -44,6 +44,13 @@ class TheThingsNetwork
     bool enableFsbChannels(int fsb);
 
   public:
+    TheThingsNetwork(fp_ttn_t *fp)
+    {
+      #ifdef TTN_FP_EU868
+        *fp = 1;
+      #endif
+      this->fp = *fp;
+    }
```

Comment:
The user provides the `fp`, so just get `fp_ttn_t fp` from the user and set it as `this->fp`. No pointers, no define checks. We can't define stuff that the library uses, unfortunately, that's not supported in Arduino.